coreneuron/utils/randoms/nrnran123.cu

```diff
@@ -127,27 +170,47 @@ CORENRN_HOST_DEVICE double nrnran123_uint2dbl(uint32_t u) {
 
 /* nrn123 streams are created from cpu launcher routine */
 void nrnran123_set_globalindex(uint32_t gix) {
-    g_k.v[0] = gix;
+    setup_global_state();
+    // If the global seed is changing then we shouldn't have any active streams.
+    {
+        std::lock_guard<OMP_Mutex> _{g_instance_count_mutex};
+        if (g_instance_count != 0) {
+            std::ostringstream oss;
+            oss << "nrnran123_set_globalindex(" << gix
+                << ") called when a non-zero number of Random123 streams (" << g_instance_count
+                << ") were active. This is not safe, some streams will remember the old value ("
+                << get_global_state().v[0] << ')';
+            throw std::runtime_error(oss.str());
+        }
+    }
```

Comment:
There wasn't any discussion, but there was a pre-existing comment:
https://github.com/BlueBrain/CoreNeuron/blob/170a0bb74b033de020946c96f67f8699e67e10d4/coreneuron/utils/randoms/nrnran123.cu#L48
which made me think that  we could easily add a diagnostic for this. I haven't tried to run the `psolve-direct` tests with this branch. You're probably right that this should be introduced as a warning instead of a hard error.

Reply:
I demoted this to a warning.